x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/schema_logic.test.ts

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogicMounter, mockFlashMessageHelpers, mockHttpValues } from '../../../__mocks__';
+import { mockEngineActions } from '../../__mocks__/engine_logic.mock';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { SchemaType, Schema } from '../../../shared/schema/types';
+
+import { SchemaLogic } from './schema_logic';
+
+describe('SchemaLogic', () => {
+  const { mount } = new LogicMounter(SchemaLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors, flashSuccessToast, setErrorMessage } = mockFlashMessageHelpers;
+
+  const MOCK_RESPONSE = {
+    schema: {
+      some_text_field: SchemaType.Text,
+      some_number_field: SchemaType.Number,
+    },
+    mostRecentIndexJob: {
+      percentageComplete: 100,
+      numDocumentsWithErrors: 10,
+      activeReindexJobId: 'some-id',
+      isActive: false,
+      hasErrors: true,
+    },
+    unconfirmedFields: ['some_field'],
+    unsearchedUnconfirmedFields: true,
+  };
+
+  const DEFAULT_VALUES = {
+    dataLoading: true,
+    schema: {},
+    isUpdating: false,
+    hasSchema: false,
+    hasSchemaChanged: false,
+    cachedSchema: {},
+    mostRecentIndexJob: {},
+    unconfirmedFields: [],
+    hasUnconfirmedFields: false,
+    hasNewUnsearchedFields: false,
+    isModalOpen: false,
+  };
+
+  /**
+   * Unfortunately, we can't mount({ schema: ... }) & have to use an action to set schema
+   * because of the separate connected logic file - our LogicMounter test helper sets context
+   * for only the currently tested file
+   */
+  const mountAndSetSchema = ({ schema, ...values }: { schema: Schema; [key: string]: unknown }) => {
+    mount(values);
+    SchemaLogic.actions.setSchema(schema);
+  };
```

Comment:
Just want to flag this as one very unfortunate hiccup in the `connect` approach to sharing state... our `LogicMounter` test helper automatically uses the current path to populate values, so we can't populate connected logics' values. So doing this:

```js
mount({ schema: {} }); // belongs to SchemaBaseLogic, not SchemaLogic
```
does nothing and doesn't actually correctly set the value. I used this as a workaround in this file for now, but not sure if we should try to come up with a better/more elegant solution or just want for `.extend()` type support 🤷

Reply:
I assume something like this:

```js
resetContext({
  defaults: {
    enterprise_search: {
      app_search: {
        some_base_logic: {
          someSharedValue: 'hello',
        },
        some_extended_logic: {
          someSpecificValue: 'world',
        },
      },
    },
  },
});
```

Looking at it that way, I can definitely see the potential of extending our test mount helper to mount with separate files, but it would be fairly tricky and add a decent amount of complexity

Reply:
Ahhh ok, that clears it up a bit for me, thanks for posting that.